pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java

```diff
@@ -0,0 +1,133 @@
+import com.linkedin.pinot.common.restlet.swagger.HttpVerb;
+import com.linkedin.pinot.common.restlet.swagger.Parameter;
+import com.linkedin.pinot.common.restlet.swagger.Paths;
+import com.linkedin.pinot.common.restlet.swagger.Summary;
+import com.linkedin.pinot.core.data.manager.offline.InstanceDataManager;
+import com.linkedin.pinot.core.data.manager.offline.SegmentDataManager;
+import com.linkedin.pinot.core.data.manager.offline.TableDataManager;
+import com.linkedin.pinot.core.indexsegment.IndexSegment;
+import com.linkedin.pinot.server.starter.ServerInstance;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.ws.rs.Produces;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.restlet.data.Status;
+import org.restlet.representation.Representation;
+import org.restlet.representation.StringRepresentation;
+import org.restlet.resource.ServerResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * API to provide table sizes
+ */
+public class TableSizeResource extends ServerResource {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TableSizeResource.class);
+
+  InstanceDataManager dataManager;
+
+  public TableSizeResource() {
+    ServerInstance serverInstance =
+        (ServerInstance) getApplication().getContext().getAttributes().get(ServerInstance.class.toString());
+    dataManager = (InstanceDataManager) serverInstance.getInstanceDataManager();
+  }
+
+  @Override
+  public Representation get() {
+    String tableName = (String) getRequest().getAttributes().get("tableName");
+    final String detailedStr = (String) getRequest().getAttributes().get("detailed");
+    return getTableSize(tableName, !"false".equalsIgnoreCase(detailedStr));
+  }
+
+  @HttpVerb("get")
+  @Description("Lists size of all the segments of the table")
+  @Summary("Show table storage size")
+  @Paths({"/table/{tableName}/size"})
+  @Produces("application/json")
+  public StringRepresentation getTableSize(
+      @Parameter(name="tableName", in = "path", description = "Table name")
+      String tableName,
+      @Parameter(name="detailed", in="query",
+          description = "true=List detailed segment sizes; false = list only table size",
+          required = false)
+      boolean detailed) {
+
+    if (dataManager == null) {
+      setStatus(Status.SERVER_ERROR_INTERNAL, "Invalid server initialization");
+      return new StringRepresentation("{\"message\" : \"Server is incorrectly initialized\"}");
+    }
+
+    TableDataManager tableDataManager = dataManager.getTableDataManager(tableName);
+    if (tableDataManager == null) {
+      setStatus(Status.CLIENT_ERROR_NOT_FOUND);
+      return new StringRepresentation("{\"message\" : \"Table " + tableName +
+          " is not found\"}");
+    }
+    TableSizeInfo tableSizeInfo = new TableSizeInfo();
+    tableSizeInfo.tableName = tableDataManager.getTableName();
```

Comment:
tableName here should have the REALTIME or OFFLINE prefix. I suppose that is how you expect it?